src/main/java/org/spongepowered/api/data/key/Key.java

```diff
@@ -66,4 +74,23 @@
      */
     DataQuery getQuery();
 
+    interface Builder<E, V extends BaseValue<E>> extends ResettableBuilder<Key<V>, Builder<E, V>> {
+
+        <T, B extends BaseValue<T>> Builder<T, B> valueElement(TypeToken<T> element, TypeToken<B> token);
```

Comment:
Just the value typetoken is sufficient, you can extract the element typetoken from the value typetoken.

Reply:
I would also name the method differently, maybe `valueType`?